snowdrop-execution/src/Snowdrop/Execution/DbActions/AVLp/Actions.hs

```diff
@@ -98,73 +108,120 @@ avlServerDbActions = fmap mkActions . newTVarIO
     mkActions var = (\recForProof ->
                         DbModifyActions
                           (mkAccessActions var recForProof)
-                          (reThrowAVLEx @k . apply var),
+                          (apply var),
                         retrieveHash var)
     mkAccessActions var recForProof = daaU
       where
         daa = DbAccessActions
                 -- adding keys to amsRequested
-                (\cA req -> liftIO $ reThrowAVLEx @k $ query cA req =<<
-                              atomically (retrieveAMS var recForProof $ AMSKeys req ))
+                (\cA req ->
+                  liftIO (atomically (retrieveAMS var recForProof $ rmap (AMSKeys . unHSetEl) req))
+                      >>= \ctx -> query ctx cA req
+                )
                 -- setting amsRequested to AMSWholeTree as iteration with
                 -- current implementation requires whole tree traversal
-                (\cA p b f -> liftIO $ reThrowAVLEx @k $ iter cA p b f =<<
-                  atomically (retrieveAMS var recForProof AMSWholeTree ))
-        daaM = DbAccessActionsM daa (\cA cs -> liftIO $ reThrowAVLEx @k $ modAccum cA cs =<< atomically (readTVar var))
+                (\cA ->
+                  liftIO (atomically (retrieveAMS var recForProof allWholeTree))
```

Comment:
Why `allWholeTree`? In fact, only a particular component shall be set to `AMSWholeTree`

Reply:
This is minor though, I think we'll need to significantly rework iteration in nearest weeks, so this can be left as a backlog issue

Reply:
no, it seems everything is correct. to build `DBIter` you need to get all avl trees and use them as change accum for iteration with corresponding prefix.

Reply:
```
type DIter chgAccum xs m = chgAccum -> m (DIter' xs m)
type DIter' xs m = Rec (IterAction m) xs
```